src/control/cmd/agent/infocache.go

```diff
@@ -132,11 +149,35 @@ func (aic *attachInfoCache) initResponseCache(resp *mgmtpb.GetAttachInfoResp, sc
 			aic.numaDeviceMarshResp[numa] = make(map[int][]byte)
 		}
 		aic.numaDeviceMarshResp[numa][len(aic.numaDeviceMarshResp[numa])] = numaDeviceMarshResp
+
+		// Any client bound to a NUMA node that has no network devices associated with it will
+		// get a response from this defaultNumaNode.
+		// This response offers a valid network device at degraded performance for those clients
+		// that need it.
+		if !haveDefaultNuma {
+			aic.defaultNumaNode = numa
+			haveDefaultNuma = true
+			aic.log.Debugf("The default NUMA node is: %d", aic.defaultNumaNode)
+		}
+
 		aic.log.Debugf("Added device %s, domain %s for NUMA %d, device number %d\n", resp.Interface, resp.Domain, numa, len(aic.numaDeviceMarshResp[numa])-1)
 	}
 
-	// If there are any entries in the cache and caching is enabled, the cache is 'initialized'
-	if len(aic.numaDeviceMarshResp) > 0 && aic.enabled.IsTrue() {
+	// If there were no network devices found, then add a default response to the default NUMA node entry
+	if _, ok := aic.numaDeviceMarshResp[aic.defaultNumaNode]; !ok {
+		aic.log.Debugf("No valid devices were detected in the fabric scan.\n" +
+			"\tThe default response contains no network interface or domain.\n" +
+			"\tCheck network configuration and run daos_agent net-scan to see network device and NUMA bindings.")
```

Comment:
I would think that this message should be at least info-level, right? While I think the suggested help is nice, it's a bit verbose. I suggest something like, "No network devices detected in fabric scan; default AttachInfo response may be incorrect." There should then be a corresponding admin guide entry with more details about what this message might mean and what corrective actions the admin might take to resolve it.

Reply:
That's reasonable.  I'll go ahead and make the change, and add a corresponding admin guide entry.